packages/backend/src/open_payments/payment/outgoing/service.ts

```diff
@@ -374,11 +392,21 @@ async function validateGrant(
     .withGraphFetched('[quote.asset]')
 
   if (grantPayments.length === 0) {
-    return true
+    return { isValid: true }
   }
 
-  let sentAmount = BigInt(0)
-  let receivedAmount = BigInt(0)
+  const amounts = {
+    sent: {
+      assetCode: payment.asset.code,
+      assetScale: payment.asset.scale,
+      value: BigInt(0)
+    },
+    received: {
+      assetCode: payment.asset.code,
+      assetScale: payment.asset.scale,
```

Comment:
I think the received amount should be the asset of the receiver itself, ie. it can be a different asset than the payment. In this case, we should use `grant.limits` to get this asset information instead.

I think debitAmount should always be the outgoing payment asset, however.

Reply:
updated: https://github.com/interledger/rafiki/pull/2753/commits/fc4adba14de448fde3d5e545aa70aac9e41e28de

I used the `paymentLimits` (comes from grant.limits) and added some typeguards. Otherwise its a bit tedious to check which of the amounts are defined.